### DISCUSSION TITLE
Serializing configuration when processing requests so can we fill servers by IServerAddressesFeature.Addresses

### DISCUSSION THREAD

**Discussion by CamusGao** [scalar/scalar#3647] [Ideas]
Serializing configuration when processing requests so can we fill servers by IServerAddressesFeature.Addresses

**Comment by xC0dex** [comment]
Hey @CamusGao,
Please provide a small code example of how the code would look where you would set the servers.
.NET's built-in generator already uses the `IServerAddressesFeature` to configure the server addresses. [Here](https://github.com/dotnet/aspnetcore/blob/main/src/OpenApi/src/Services/OpenApiDocumentService.cs#L195) is the code.
